src/Mod/BIM/bimcommands/BimWindow.py

```diff
@@ -254,6 +286,37 @@ def update(self,point,info):
             delta = FreeCAD.Rotation(r[0],r[1],r[2],r[3]).multVec(FreeCAD.Vector(delta.x,-delta.y,-delta.z))
         self.tracker.pos(point.add(delta))
 
+    def getLibraryPresetPath(self):
+        p = FreeCAD.ParamGet("User parameter:Plugins/parts_library").GetString("destination", "")
+        # librarypath should have only forward slashes already, but let's use replace() anyway just to be sure:
+        return p.replace("\\", "/") + "/Architectural Parts"
+
+    def scanLibraryPresets(self):
+        # because of the use of FreeCADGui.doCommand() backslashes in the
+        # paths in librarypresets need to be double escaped "\\\\", so let's
+        # use forward slashes instead...
+
+        presets = []
+        librarypath = self.getLibraryPresetPath()
+        presetdir = FreeCAD.getUserAppDataDir().replace("\\", "/") + "/Arch"
+        for path in [librarypath, presetdir]:
+            if not os.path.isdir(path):
+                continue
+            for wtype in ["Windows", "Doors"]:
+                wdir = path + "/" + wtype
+                if not os.path.isdir(wdir):
+                    continue
+                for subtype in os.listdir(wdir):
+                    subdir = wdir + "/" + subtype
+                    if not os.path.isdir(subdir):
+                        continue
+                for subfile in os.listdir(subdir):
+                    if (os.path.isfile(subdir + "/" + subfile) and subfile.lower().endswith(".fcstd")):
+                        p1 = wtype + " - " + subtype + " - " + subfile[:-6]
+                        p2 = subdir + "/" + subfile
+                        presets.append(p1, p2)
```

Comment:
```suggestion
                        presets.append([p1, p2])
```
Typo? From a quick glance at Yorik's comment about the error he was getting.